src/complex.jl

```diff
@@ -214,26 +214,64 @@ end
 # Also give complex-valued degree definitions. We choose not to overwrite degree, as this will lead to issues in monovecs
 # and their sorting. So now there are two ways to calculate degrees: strictly by considering all variables independently,
 # and also by looking at their complex structure.
+for fn in (:degree_complex, :halfdegree)
+    @eval function $fn(t::AbstractTermLike)
+        realdeg = 0
+        cpdeg = 0
+        conjdeg = 0
+        for (var, exp) in powers(t)
+            if isreal(var)
+                realdeg += exp
+                (isrealpart(var) || isimagpart(var)) && error(
+                    "Cannot calculate complex degrees when real or imaginary parts are present",
+                )
+            else
+                if isconj(var)
+                    conjdeg += exp
+                else
+                    cpdeg += exp
+                end
+            end
+        end
+        return $(
+            fn === :degree_complex ? :(realdeg) : :(div(realdeg, 2, RoundUp))
+        ) + max(cpdeg, conjdeg)
+    end
+end
+
 """
     degree_complex(t::AbstractTermLike)
 
 Return the _total complex degree_ of the monomial of the term `t`, i.e., the maximum of the total degree of the declared
 variables in `t` and the total degree of the conjugate variables in `t`.
 To be well-defined, the monomial must not contain real parts or imaginary parts of variables.
+If `x` is a real-valued variable and `z` is complex-valued,
+- `degree_complex(x^5) = 5`
+- `degree_complex(z^3 * conj(z)^4) = 4` and `degree_complex(z^4 * conj(z)^3) = 4`
```

Comment:
I meant also an example with two different complex-valued where one has a larger degree for the conjugate and not the other one

Reply:
Ok, now I use two real and two complex variables for the example, I hope this covers what you thought of.